src/overlays/actors/ovl_Bg_Mori_Bigst/z_bg_mori_bigst.c

```diff
@@ -20,39 +36,219 @@ const ActorInit Bg_Mori_Bigst_InitVars = {
+        ((this->dyna.actor.initPosRot.pos.y - 5.0f) <= PLAYER->actor.posRot.pos.y)) {
+        BgMoriBigst_SetupFall(this, globalCtx);
+        func_800800F8(globalCtx, 0xC94, 0x48, &this->dyna.actor, 0);
+    }
+}
+
+void BgMoriBigst_SetupFall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_Fall);
+}
+
+void BgMoriBigst_Fall(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor_MoveForward(&this->dyna.actor);
+    if (this->dyna.actor.posRot.pos.y <= this->dyna.actor.initPosRot.pos.y) {
+        this->dyna.actor.posRot.pos.y = this->dyna.actor.initPosRot.pos.y;
+        BgMoriBigst_Quake(this, globalCtx);
+        Audio_PlayActorSound2(&this->dyna.actor, NA_SE_EV_STONE_BOUND);
+        func_800800F8(globalCtx, 0x3FC, 8, &this->dyna.actor, 0);
+        func_8002DF38(globalCtx, NULL, 0x3C);
+    }
+}
+
+void BgMoriBigst_Quake(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Camera** cameras = globalCtx->cameraPtrs;
+    s32 quake;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_SetupSpawnStalfosPair);
+    this->waitTimer = 18;
+    quake = Quake_Add(cameras[globalCtx->activeCamera], 3);
+    Quake_SetSpeed(quake, 0x61A8);
+    Quake_SetQuakeValues(quake, 5, 0, 0, 0);
+    Quake_SetCountdown(quake, 16);
+}
+
+void BgMoriBigst_SetupSpawnStalfosPair(BgMoriBigst* this, GlobalContext* globalCtx) {
+    if (this->waitTimer <= 0) {
+        BgMoriBigst_SpawnStalfosPair(this, globalCtx);
+    }
+}
+
+void BgMoriBigst_SpawnStalfosPair(BgMoriBigst* this, GlobalContext* globalCtx) {
+    Actor* stalfos1;
+    Actor* stalfos2;
+
+    BgMoriBigst_SetActionFunction(this, BgMoriBigst_StalfosPairWait);
+    Flags_UnsetClear(globalCtx, this->dyna.actor.room);
+    stalfos1 =
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 2, 70.0f, 827.0f, -3383.0f, 0, 0, 0, 5);
+    if (stalfos1 != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Warning: 3-1 Stalfos failure
+        osSyncPrintf("Warning : 第３-1スタルフォス発生失敗\n");
+    }
+    stalfos2 =
+        Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, 2, 170.0f, 827.0f, -3260.0f, 0, 0, 0, 5);
+    if (stalfos2 != NULL) {
+        this->dyna.actor.child = NULL;
+        this->dyna.actor.initPosRot.rot.z++;
+    } else {
+        // Warning: 3-2 Stalfos failure
+        osSyncPrintf("Warning : 第３-2スタルフォス発生失敗\n");
+    }
+    Flags_SetClear(globalCtx, this->dyna.actor.room);
+}
+
+void BgMoriBigst_StalfosPairWait(BgMoriBigst* this, GlobalContext* globalCtx) {
+    if ((this->dyna.actor.initPosRot.rot.z == 0) && !func_8008E988(globalCtx)) {
+        Flags_SetSwitch(globalCtx, (this->dyna.actor.params >> 8) & 0x3F);
+        BgMoriBigst_FinalNoop(this, globalCtx);
+    }
+}
+
+void BgMoriBigst_FinalNoop(BgMoriBigst* this, GlobalContext* globalCtx) {
+    BgMoriBigst_SetActionFunction(this, NULL);
+}
```

Comment:
Should this be named like `SetupNoop` since it sets up a noop rather than being a noop (I also think `Idle` is better than `Noop` but idk)

Reply:
Noop is apparently the standard. The actor has two different SetupNoop functions, one at the start if certain conditions are met and one at the end when it's done doing things, so I gave them different names. Open to suggestions.